src/XMakeBuildEngine/BackEnd/Components/Logging/LoggingService.cs

```diff
@@ -260,6 +262,19 @@ protected LoggingService(LoggerMode loggerMode, int nodeId)
             }
 
             _serviceState = LoggingServiceState.Instantiated;
+
+            TelemetryPipelines = new List<ITelemetryPipeline>
```

Comment:
This is the part of the code where it needs to dynamically discover pipelines, we'll discuss this soon.